### DISCUSSION TITLE
Tali AI Grant Proposal - Technical Support Bot/Widget For Akash Network

### DISCUSSION THREAD

**Discussion by OlypsisAli** [akash-network/community#237] [AI - Artificial Intelligence]
### Introducing Tali AI ([tryTali.com](http://trytali.com/))

We propose to build a specialized Discord bot (and or widget that lives on docs page) for Akash that will use the latest technology around LLM’s (Large Language Models) to import Akash's Github repos, documentation, video content and Discord Q&A’s to provide an interface for Akash users/developers to ask questions in natural language.

The idea is to provide another member to Akash's DevRel team, reducing support hours on Discord and a path of triage for commonly asked questions (making the team and users get answers quicker).

### Background

The core of technical Web3 projects depends on a healthy ecosystem of developers using the protocols. Developers learn how to use the various projects through inspecting developer docs and Github repositories. As expected, the documentation doesn’t cover everything and a team of people are needed to provide “front line” support to those building / using the protocol.

Enter Developer Relations, a team tasked with being the first human interface with protocols who (according to [Chainlink](https://blog.chain.link/why-devrel-crucial-to-success-web3/)) provide 2 critical roles the project: building developer communities & tracking product market fit.

The issue that often arises is a lack of human capital to be able to deal with the number of requests & questions that populate a DevRel teams day.

We propose to provide a solution that will enable the Akash DevRel team to be able to focus on what they do best: solving unique customer problems and building trust in the protocol. Providing a bot that can answer previously asked questions will enable DevRel to become more effective and in turn provide higher-quality support to end users.

### Team 

Ali Agha

Ali Agha is a technologist and entrepreneur with a focus on decentralized solutions. With his previous venture, Olypsis Technologies, Ali provided Web3 consulting services for countless startups and major companies like IBM and Thomson Reuters. Ali first became engaged in the blockchain space in 2015 when he discovered bitcoin. Since that time he has dedicated his career to creating a more fair and just world through the power of decentralization.

Github: https://github.com/OlypsisAli
Twitter: https://twitter.com/iamAliAgha

Tenzin Rose

Tenzin Rose is an entrepreneur and full-stack developer with a background in cloud infrastructure sales. He’s worked with startups/enterprises globally, helping them deploy projects successfully & drive revenue. His current passions are web development, all things Ethereum and getting to grips with the moon math in ZKP.

He’s worked previously on many interesting projects including building out zkGiving, an app that enables people to donate privately to charities of their choice.

Github: https://github.com/niznet89
Projects: https://www.m00nlink.xyz/tenzin
Twitter: https://twitter.com/tenzin_rose

### Project Plan 

**Pre Implementation:** We’d like to work with the DevRel team to understand their needs, sticking points and time sinks to see where the bot can be of most assistance. We’ll use these insights to see which data sources need to be implemented.

**Milestone 1:** Implementation of Data Sources [2 weeks]: Once discovery has been done, we’ll spend the time implementing the 3-4 most high value data sources to Tali. We will leverage a vector database to enable embeddings based search on queries. Optimization around how files are loaded, index creation and integrating various data sources will need to be tested.

**End state:** Have 3-4 data sources that can be queried on. Expected bugs: hallucinations, mismatch on query / documents.

**Milestone 2:** Testing + Optimization [2 weeks]: Arguably the most important part of the process. The focus will be to discard hallucinations and optimize prompt engineering so only relevant answers appear, ideally with source material & links for the user to follow up on. The purpose of this will be to make the bot production ready.

**End state:** Production-ready bot ready (once signed off on by DevRel team) to be deployed to Akash discord and or webpage via a widget.

### Funding/Budget

Per the grant, we would like to apply for a $5K USD grant to build and maintain the bot. This will cover our time, hosting, LLM token costs, and future improvements for one year.

### Contact

Email: [ali@trytali.com](mailto:ali@trytali.com)
Twitter: @iamAliAgha
Github: @OlypsisAli

**Comment by bmenzalji** [comment]
Do you have a working proof of concept for this for any project?  If so, can you provide a link to it?
